estuary/ui
# Track instances of unsupported connector versions in edit workflow

## Issues

The issues directly below are completely resolved by this PR:
https://github.com/estuary/ui/issues/941

## Changes

### 941

The following features are included in this PR:

* Use the schemas of latest connector version to render the forms in the edit workflow for a task using an unsupported connector version. The connector version used by the specification should not change.

-   Add a log rocket event to track instances of unsupported connector versions in edit workflows.

## Tests

### Manually tested

Approaches to testing are as follows:

* Capture Tests:

    * Enter the capture create workflow, select a source connector which has multiple supported versions, complete the form, publish the capture, and observe that the capture uses the latest version of the source connector.

    * Enter the capture create workflow, select a source connector which has multiple supported versions, change the selected connector in the _Details_ section of the form, and observe that the drafted specification uses the latest version of the whichever connector is selected.

    * Enter the capture create workflow, select a source connector which has multiple supported versions, toggle the selected connector in the _Details_ section of the form (i.e., select a different connector then return to the initial connector selected), and observe that the drafted specification uses the latest version of the whichever connector is selected.

    * Enter the edit workflow of a capture that uses an older version of a source connector, edit the resource configuration of an existing binding, observe that a LogRocket event was fired including the existing and evaluated connector tag IDs, edit the endpoint configuration, publish the altered specification, and observe that the version of the source connector was preserved.

### Automated tests

N/A

## Screenshots

N/A
